### PR TITLE
tests: mutex_order: updated test and fixed README

### PR DESCRIPTION
**Pull request by OlegHahm**
The README of this test is incorrect. Furthermore, the test includes an unnecessary dependency to `xtimer`. As a bonus feature this PR also includes an according pexpect script for this test.

### CODE REVIEW

File: tests/mutex_order/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+thread_prio = {
+        3:  6,
+        4:  4,
+        5:  0,
+        6:  2,
+        7:  1
```
Comment (miri64) [inline tests/mutex_order/tests/01-run.py]: To dependend on the C code IMHO, but let's get this merged ASAP.

**Comment by miri64** [comment]
I would recommend to only provide a backport after #6050 was merged so you don't have to rebase ;-)

**Comment by A-Paul** [comment]
Output and README.md correlate now. The python script isn't very helpful as it is. A statement "SUCCESS/FAILED" is missing. Also, `make test` doesn't exist.
But that's not an error and can be improved later.

**Comment by miri64** [comment]
@A-Paul pexpect fails with an exception if the output is wrong, so you definitely will notice ;-)

**Comment by miri64** [comment]
Murdock fail due to #6036. Go.

**Comment by miri64** [comment]
Please provide a backport, I want to have an RC tomorrow morning (preferably with this and a backport of #6053).

**Comment by A-Paul** [comment]
> @A-Paul pexpect fails with an exception if the output is wrong, so you definitely will notice ;-)

For a moment I was really afraid that you are serious about that.

**Comment by miri64** [comment]
Backport provided in #6059

**Pull request merged**